ingestion/src/metadata/profiler/metrics/static/mean.py

```diff
@@ -74,21 +74,35 @@ def fn(self):
         return None
 
     # pylint: disable=import-outside-toplevel
-    def df_fn(self, df=None):
+    def df_fn(self, dfs=None):
         """dataframe function"""
-        from numpy import vectorize
-        from pandas import DataFrame
+        import pandas as pd
+        from numpy import average, vectorize
 
-        df = cast(DataFrame, df)  # satisfy mypy
+        dfs = cast(List[pd.DataFrame], dfs)
+
+        means = []
+        weights = []
 
         if is_quantifiable(self.col.type):
-            return df[self.col.name].mean()
+            for df in dfs:
+                mean = df[self.col.name].mean()
+                if not pd.isnull(mean):
+                    means.append(mean)
+                    weights.append(df[self.col.name].count())
 
         if is_concatenable(self.col.type):
-            length_vector_fn = vectorize(len)
-            return length_vector_fn(df[self.col.name]).mean()
+            length_vectorize_func = vectorize(len)
+            for df in dfs:
+                mean = length_vectorize_func(df[self.col.name].dropna()).mean()
```

Comment:
should we add `.astype(str)` here as a safe guard?

Reply:
yes, that would resolve some edge cases i was facing